cxgo/main.go

```diff
@@ -319,6 +322,35 @@ func main() {
 	registerFlags(&options)
 	flag.Parse()
 
+	if len(options.configPath) > 0 {
+		if !strings.HasSuffix(options.configPath, ".json") {
+			panic(errors.New("config file path is not pointing to json file"))
+		}
+		if len(options.configHash) > 0 {
+			if err := cxTracker.GetConfigFromTrackerService(options.configHash, options.configPath); err != nil {
+				panic(errors.New(fmt.Sprintln("unable to fetch config from remote service locally due to error %v", err)))
+			}
+		}
+
+		file, err := ioutil.ReadFile(options.configPath)
+		if err != nil {
+			panic(errors.New(fmt.Sprintln("config file can't be read from path %v due to error %v", options.configPath, err)))
+		}
+		configFromFile := cxtracker.CXApplicationConfig{}
+		if err := json.Unmarshal([]byte(file), &configFromFile); err != nil {
+			panic(errors.New(fmt.Sprintln("config from path %v can't be unmarshalled to JSON struct due to error %v", options.configPath, err)))
+		}
+
+		// TODO we probably need to do few more steps here like creating the wallet and
```

Comment:
Here are two points to be discussed - should wallet be created automatically and those configs updated.

As well as second part, there is no point in running with --blockchain flag as far as I understand this topic.

Reply:
Additional thing to discuss is - should potential command line arguments overwrite config value?
Right now config has highest priority and I'm not sure is it even a valid use case to consider mixing those two?

Reply:
Automatic creation of wallets could be good. Wallets don't really make sense for CX chains, as we're managing programs, not coins. Wallets are used to sign transactions, but I don't see any reason to have several wallets.

You need to run with `--blockchain` if you need to generate a genesis signature.

Reply:
Most projects I've used usually prioritize command-line arguments. And, at least to me, this makes sense: if a user is explicitly adding a command line argument at the moment of wanting to run your application, then it means that the user really wants that configuration setting to be the one to be used. Environment variables should be the default, then config files should override these, and then command-line arguments should override everything.